streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -194,19 +129,28 @@ public StreamTask(final TaskId id,
         this.time = time;
         this.producerSupplier = producerSupplier;
         this.producer = producerSupplier.get();
-        final String threadId = Thread.currentThread().getName();
-        this.taskMetrics = new TaskMetrics(threadId, id, streamsMetrics);
+        this.streamsMetrics = streamsMetrics;
 
+        threadId = Thread.currentThread().getName();
+        final String taskId = id.toString();
         closeTaskSensor = ThreadMetrics.closeTaskSensor(threadId, streamsMetrics);
+        final Sensor parent = ThreadMetrics.commitOverTasksSensor(threadId, streamsMetrics);
+        commitSensor = TaskMetrics.commitSensor(threadId, taskId, streamsMetrics, parent);
+        enforcedProcessingSensor = TaskMetrics.enforcedProcessingSensor(threadId, taskId, streamsMetrics, parent);
+        processLatencySensor = TaskMetrics.processLatencySensor(threadId, taskId, streamsMetrics);
+        punctuateLatencySensor = TaskMetrics.punctuateSensor(threadId, taskId, streamsMetrics);
+        recordLatenessSensor = TaskMetrics.recordLatenessSensor(threadId, taskId, streamsMetrics);
+        TaskMetrics.droppedRecordsSensor(threadId, taskId, streamsMetrics);
```

Comment:
Nice cleanup, it makes looking at `StreamTask` bit easier having `TaskMetrics` in a separate class.